src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -242,33 +257,61 @@ async Task<JObject> ResolveAsLocalOrThisMember(string name)
                     return null;
                 }
 
-                JToken objRet = valueOrError.Value.FirstOrDefault(objPropAttr => objPropAttr["name"].Value<string>() == nameTrimmed);
+                JToken objRet = valueOrError.Value.FirstOrDefault(objPropAttr => objPropAttr["name"].Value<string>() == name);
                 if (objRet != null)
                     return await GetValueFromObject(objRet, token);
 
                 return null;
             }
 
-            async Task<JObject> ResolveAsInstanceMember(string expr, JObject baseObject)
+            async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject baseObject, bool hasNullCondition)
             {
+                // if the prevPart has "?" at the end: true
+                bool hasResolvedObjNullCondition = hasNullCondition;
+                bool isPrevPartNull = false;
                 JObject resolvedObject = baseObject;
-                string[] parts = expr.Split('.');
-                for (int i = 0; i < parts.Length; i++)
+                for (int i = 0; i < parts.Count; i++)
                 {
-                    string partTrimmed = parts[i].Trim();
+                    string partTrimmed = parts[i];
                     if (partTrimmed.Length == 0)
                         return null;
 
+                    // current value of resolvedObject is on parts[i - 1]
+                    if (isPrevPartNull || resolvedObject.IsNullValuedObject())
+                    {
+                        // trying null.$member
+                        if (!hasResolvedObjNullCondition)
+                            throw new ReturnAsErrorException($"Cannot access member \"{partTrimmed}\" of a null-valued object.", "ReferenceError");
+
+                        if (i == parts.Count - 1)
+                        {
+                            // this is not ideal, it returns the last object
+                            // that had objectId and was null-valued,
+                            // so the class/description of object are not of the last part
+                            return resolvedObject;
+                        }
+                        // else: trying null?.$member... - allowed operation
+                        isPrevPartNull = true;
+                    }
+
+                    bool hasCurrentPartNullCondition = partTrimmed.Last() == '?';
+                    hasResolvedObjNullCondition = hasCurrentPartNullCondition;
+
+                    // cannot resolve the members but can check if null condition is
+                    // correctly applied and if we should throw or return null-object
                     if (!DotnetObjectId.TryParse(resolvedObject?["objectId"]?.Value<string>(), out DotnetObjectId objectId))
-                        return null;
+                        continue;
```

Comment:
You can check here if `resolvedObject` was a primitive type (except `string`), so maybe check `type == 'number'`, and throw if that's the case.

Reply:
By throwing here this way we will block executing methods on primitives. I will check out if there's any other solution.

Reply:
What's more I decided to leave the `objectId` check, so not exchanging it to `type` value comparison. We want strings to behave the same was as other primitives at this stage because we don't support evaluation of their properties yet (it is planned to be solved in https://github.com/dotnet/runtime/issues/66823). Till this PR is fixed I will throw "not supported" error for strings so that users won't get confused.